src/main/java/org/commonjava/storage/pathmapped/datastax/CassandraPathDB.java

```diff
@@ -73,21 +86,29 @@ public CassandraPathDB( PathMappedStorageConfig config )
         session.execute( getSchemaCreateTablePathmap( keyspace ) );
         session.execute( getSchemaCreateTableReversemap( keyspace ) );
         session.execute( getSchemaCreateTableReclaim( keyspace ) );
+        session.execute( getSchemaCreateTableFileChecksum( keyspace ) );
 
         MappingManager manager = new MappingManager( session );
 
         pathMapMapper = manager.mapper( DtxPathMap.class, keyspace );
         reverseMapMapper = manager.mapper( DtxReverseMap.class, keyspace );
         reclaimMapper = manager.mapper( DtxReclaim.class, keyspace );
 
-        preparedSingleExistQuery = session.prepare( "SELECT count(*) FROM " + keyspace
-                                                                    + ".pathmap WHERE filesystem=? and parentpath=? and filename=?;" );
+        preparedSingleExistQuery = session.prepare(
+                "SELECT count(*) FROM " + keyspace + ".pathmap WHERE filesystem=? and parentpath=? and filename=?;" );
 
         preparedDoubleExistQuery = session.prepare( "SELECT count(*) FROM " + keyspace
-                                                                    + ".pathmap WHERE filesystem=? and parentpath=? and filename in (?,?);" );
+                                                            + ".pathmap WHERE filesystem=? and parentpath=? and filename in (?,?);" );
+
+        preparedListQuery =
+                session.prepare( "SELECT * FROM " + keyspace + ".pathmap WHERE filesystem=? and parentpath=?;" );
+
+        preparedPathMapByIdQuery =
```

Comment:
Aha, this is what we should avoid. The "ALLOW FILTERING" could be a big performance killer. In Cassandra, this only should be used when "you know and can live with the performance penalty". In other words, if a table is small-sized and the cluster nodes are less and so on, use it after testing the performance. Basically, we should only query based on the "cluster key" fields (the fields in the PRIMARY key following the partition key). We can solve this problem in another way. I will comment it below on the insert code.

Reply:
Ok, that means we need to do redundancy for the field we need to do filtering.  Maybe just as said as above, a storagepath field is needed in separate table.